### PR TITLE
GG-27792 Added fast-check mode to Partition reconciliation task.

### PR DESCRIPTION
**Pull request by sk0x50**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java [hunk 1]
```diff
@@ -730,6 +731,33 @@ public GridDhtPartitionExchangeId exchangeId() {
         return exchId;
     }
 
+    /**
+     * Returns a collection of partitions (per cache group) that did not pass validation by update counter or size.
+     *
+     * Please take into account that the following cache configurations disable partitions validation:
```
Comment (sanpwc) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java]: Could you please rephrase it?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionStateValidationException.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+package org.apache.ignite.internal.processors.cache.distributed.dht.topology;
+
+import java.util.Map;
+import java.util.Set;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.processors.affinity.AffinityTopologyVersion;
+import org.apache.ignite.internal.processors.cache.distributed.dht.preloader.GridDhtPartitionsExchangeFuture;
+
+/**
+ * Yhis exception represents a result of validation partitions for a cache group.
```
Comment (sanpwc) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionStateValidationException.java]: typo

File: modules/core/src/main/java/org/apache/ignite/internal/visor/checker/VisorPartitionReconciliationTaskArg.java [hunk 3]
```diff
@@ -42,6 +41,16 @@
     /** Print result to locOutput. */
     private boolean locOutput;
 
+    /** Flag indicates that only subset of partitions should be checked and repaired. */
+    private boolean fastCheck;
+
+    /**
+     * Collection of partitions whcih should be validated.
```
Comment (sanpwc) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/checker/VisorPartitionReconciliationTaskArg.java]: typo

File: modules/core/src/main/java/org/apache/ignite/internal/visor/checker/VisorPartitionReconciliationTaskArg.java [hunk 4]
```diff
@@ -95,6 +148,8 @@ public VisorPartitionReconciliationTaskArg(Set<String> caches, boolean repair, b
 
         out.writeBoolean(includeSensitive);
 
+        out.writeBoolean(fastCheck);
```
Comment (sanpwc) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/checker/VisorPartitionReconciliationTaskArg.java]: Compatibility issue.

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/cache/PartitionReconciliation.java [hunk 5]
```diff
@@ -210,7 +221,8 @@ private ReconciliationResult partitionReconciliationCheck(
     @Override public void parseArguments(CommandArgIterator argIter) {
         Set<String> cacheNames = null;
         boolean repair = false;
-        boolean verbose = (boolean)INCLUDE_SENSITIVE.defaultValue();
+        boolean fastCheck = false;
```
Comment (sanpwc) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/cache/PartitionReconciliation.java]: Why not to use .defaultValue()?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java (hunk 1 above)
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java]: I changed in the following way:
`Partitions are not validated in the following cases and therefore they will not be returned by this method`

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/cache/PartitionReconciliation.java (hunk 5 above)
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/internal/commandline/cache/PartitionReconciliation.java]: You are right. Will do.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/checker/VisorPartitionReconciliationTaskArg.java (hunk 3 above)
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/checker/VisorPartitionReconciliationTaskArg.java]: Thanks! Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/checker/VisorPartitionReconciliationTaskArg.java (hunk 4 above)
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/checker/VisorPartitionReconciliationTaskArg.java]: Nice catch! Will fix.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionStateValidationException.java (hunk 2 above)
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionStateValidationException.java]: Thanks! Fixed.

**Pull request closed without merging**